### DISCUSSION TITLE
Linux...

### DISCUSSION THREAD

**Discussion by SmRzT** [YimMenu/YimMenu#3485] [Q&A] [answered]
So i haven't played gta in a few months and it seems in recent updates they have patched i believe the asi loader method of injecting yim.

This method being using using [UltimateASILoader](https://github.com/ThirteenAG/Ultimate-ASI-Loader/releases) and renaming bink2w64.dll to bink2w64Hooked.dll but it would be that gta no longer boots if that file is renamed to hooked and only boots with the original. _I have tried renaming it to other things like dinput8 and dsound and tried xenos but it wont load and xenos crashes it_

if anyone knows of another method, would be greatly appreciated :D

**Arch Wayland**
**Heroic Launcher (can download on other stuff if need)**

**Comment by tupoy-ya** [comment, marked as answer]
> This method being using using [UltimateASILoader](https://github.com/ThirteenAG/Ultimate-ASI-Loader/releases) and renaming bink2w64.dll to bink2w64Hooked.dll but it would be that gta no longer boots if that file is renamed to hooked and only boots with the original. I have tried renaming it to other things like dinput8 and dsound and tried xenos but it wont load and xenos crashes it

No idea what's wrong, [UltimateASILoader](https://github.com/ThirteenAG/Ultimate-ASI-Loader/releases) works fine for me. I renamed the games `bink2w64.dll` to `bink2w64Hooked.dll` then downloaded `bink2w64.dll` from releases and put it in the game directory.

Maybe an update changed something in `bink2w64.dll` and you need to make a new `bink2w64Hooked.dll`, unlikely but possible.
Also don't hijack `dinput8.dll` as it is the only dll R* checks I believe.

You can also try using [YimASI](https://github.com/maybegreat48/YimASI), a pain to compile on Linux but it works.
To use it you will need to open winecfg and add a dll override for `WTSAPI32.dll`.

**Reply by SmRzT** [reply]
So it seems there was a problem with my bink2w64 file, which tbh is confusing cause i reinstalled the game yesturday fresh, but i deleted it and repaired the game and it now boots with Yim, Tysm :D

and will also keep in mind YimASI if it goofs again.
also thanks for the info on dinput8 idk bout dat. :D

**Comment by FIying-Scotsman** [comment]
> So i haven't played gta in a few months and it seems in recent updates they have patched i believe the asi loader method of injecting yim.
> 
> This method being using using [UltimateASILoader](https://github.com/ThirteenAG/Ultimate-ASI-Loader/releases) and renaming bink2w64.dll to bink2w64Hooked.dll but it would be that gta no longer boots if that file is renamed to hooked and only boots with the original. _I have tried renaming it to other things like dinput8 and dsound and tried xenos but it wont load and xenos crashes it_
> 
> if anyone knows of another method, would be greatly appreciated :D
> 
> **Arch Wayland** **Heroic Launcher (can download on other stuff if need)**

You have to keep in mind this is exploiting DLL load order hijacking to function (i.e the game looks for a DLL by name, finds yours instead of the 'real' one and loads that. Your DLL then loads the 'real' one to prevent the process crashing). 

As you're running it via WINE (I assume as you're on Linux) - this may not work properly or have been fixed in a later version of WINE as it's technically a bug in the OS. It could be WINE loading your modified DLL but then isn't loading the real one in turn - hence the crash/game not loading.

Xenos won't inject that DLL as it's designed to be a proxy itself (i.e the process is meant to load the proxy, not have it injected separately) - not a standalone executable library. You should be able to use Xenos to inject Yim or other ASI files however as ASI files are just DLLs with a different extension.

**Reply by tupoy-ya** [reply]
> This may not work properly or have been fixed in a later version of WINE as it's technically a bug in the OS. It could be WINE loading your modified DLL but then isn't loading the real one in turn - hence the crash/game not loading.

This "bug" is kind of fixed in wine, but you can unfix it by adding a DLL override in winecfg.
Also `bink2w64.dll` is not a system library so wine will load it from the game directory without any configuration.

**Comment by snowleopardd** [comment]
ppl who have a decent pc and use linux should be in jail lmao
